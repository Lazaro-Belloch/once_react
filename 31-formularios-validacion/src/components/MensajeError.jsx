import React from 'react'


const MensajeError = (props) => {

    if (!props.mensaje) {
        return null;
    }

    return (
        <div className="ui negative message">
            <div className="header">
                {props.mensaje}
            </div>
        </div>
    )
}

export default MensajeError