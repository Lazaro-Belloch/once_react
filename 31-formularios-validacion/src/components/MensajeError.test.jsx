import MensajeError from './MensajeError';
import renderer from 'react-test-renderer'
import React from 'react'


test('should not render on null', () => {
    const mensaje = renderer.create(
        <MensajeError></MensajeError>
    ).toJSON();
    expect(mensaje).toBeNull();
})

test('should render the message', () => {
    const mensajeTree = renderer.create(
        <MensajeError mensaje="kkk"></MensajeError>, 
    ).toJSON();
    expect(mensajeTree).not.toBeNull()
    expect(mensajeTree).toMatchSnapshot()

})

