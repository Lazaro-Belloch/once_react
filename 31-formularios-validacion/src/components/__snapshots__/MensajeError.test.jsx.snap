// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render the message 1`] = `
<div
  className="ui negative message"
>
  <div
    className="header"
  >
    kkk
  </div>
</div>
`;
